core/livepeernode.go

```diff
@@ -62,23 +71,65 @@ func NewLivepeerNode(e eth.LivepeerEthClient, vn net.VideoNetwork, nodeId NodeID
 		return nil, ErrLivepeerNode
 	}
 
-	return &LivepeerNode{VideoDB: NewVideoDB(vn.GetNodeID()), VideoNetwork: vn, Identity: nodeId, Addrs: addrs, Eth: e, WorkDir: wd}, nil
+	return &LivepeerNode{VideoDB: NewVideoDB(vn.GetNodeID()), VideoNetwork: vn, Identity: nodeId, Addrs: addrs, Eth: e, WorkDir: wd, PeerConns: make([]PeerConn, 0)}, nil
 }
 
 //Start sets up the Livepeer protocol and connects the node to the network
-func (n *LivepeerNode) Start(bootID, bootAddr string) error {
+func (n *LivepeerNode) Start(ctx context.Context, bootID, bootAddr string) error {
 	//Set up protocol (to handle incoming streams)
 	if err := n.VideoNetwork.SetupProtocol(); err != nil {
 		glog.Errorf("Error setting up protocol: %v", err)
 		return err
 	}
 
 	//Connect to bootstrap node.  This currently also kicks off a bootstrap process, which periodically checks for new peers and connect to them.
-	if err := n.VideoNetwork.Connect(bootID, bootAddr); err != nil {
-		glog.Errorf("Cannot connect to node: %v", err)
-		return err
+	if bootID != "" && bootAddr != "" {
+		if err := n.VideoNetwork.Connect(bootID, bootAddr); err != nil {
+			glog.Errorf("Cannot connect to node: %v", err)
+		} else {
+			n.PeerConns = append(n.PeerConns, PeerConn{NodeID: bootID, NodeAddr: bootAddr})
+		}
 	}
 
+	//Load connection information, try and connect to them.
+	bytes, err := ioutil.ReadFile(fmt.Sprintf("%v/conn", n.WorkDir))
+	if err != nil {
+		for _, line := range strings.Split(string(bytes), "\n") {
+			larr := strings.Split(line, "|")
+			if len(larr) == 2 {
+				if err := n.VideoNetwork.Connect(larr[0], larr[1]); err != nil {
+					glog.Errorf("Cannot connect to node: %v", err)
```

Comment:
Hmm, I'm not sure this is an error. It's just the fact that the remote node wasn't online anymore.

Reply:
We should also probably remove the node from PeerConns at this point. Otherwise it will grow indefinitely.

Reply:
Yeah - I agree.  `Start` is only called once, so this is essentially "initializing" `PeerConns`.